### DISCUSSION TITLE
Difference between Apply and Alter?

### DISCUSSION THREAD

**Discussion by devlife** [OrchardCMS/OrchardCore#13421] [Q&A] [answered]
What is the difference between `ContentItemExtensions.Apply()` and `ContentItemExtensions.Alter()`?

**Comment by hishamco** [comment, marked as answer]
`Alter()` can create or modifies the content item, while `Apply()` updates the content item. That's why you will find that `Apply()` used internally by `Alter()`

Seems that's why I saw `Alter()` has been used on almost the migrations

**Reply by devlife** [reply]
Thanks very much!!
